pkg/storage/engine/sst_writer.go

```diff
@@ -178,7 +179,7 @@ func (fw *SSTWriter) Close() {
 // Implements the writeCloseSyncer interface and is intended for use with
 // SSTWriter.
 type MemFile struct {
-	data []byte
+	bytes.Buffer
```

Comment:
Random drive by comment: embedding it this way inherits all https://golang.org/pkg/bytes/#Buffer methods, which I don't believe is something we want.